lib/evm/src/client.rs

```diff
@@ -43,7 +43,13 @@ impl EVMClient for BasinClient {
             .create_pub(owner, name.into())
             .send()
             .await
-            .map_err(|e| Error::Evm(e.to_string()))?;
+            .map_err(|e| {
+                if e.is_revert() {
```

Comment:
I've tried to use `e.decode_contract_revert()` to decode the error and check the revert message, but could not get it to work. So I've settled on just checking if it's a revert. Not ideal, but good enough for now.

Reply:
`e.decode_contract_revert()` did it return `None`?

Reply:
> `e.decode_contract_revert()` did it return `None`?

yeah! not sure why